prometheus/metrics.go

```diff
@@ -12,7 +13,9 @@ import (
 	"github.com/prometheus/common/model"
 
 	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
```

Comment:
Having the prometheus client depending on the k8s client is somewhat annoying ... they are both meant to be kind of 'low level' parts of kiali, hence autonomous.

Would it be possible to move the related parts to an upper level?
For instance, iiuc the kubernetes client is mostly needed to adjust the query range low bound, and in `getMetrics` we have the `q *MetricsQuery` object that contains the range, so this object could be updated before being sent to the prom client.
This way, prom client still is agnostic of any k8s consideration.

Another advantage is that it would allow some optimization, if metrics are fetched several times for the same namespace, to fetch the namespace creation time only once.

Reply:
Note also that in `handlers/metrics.go`, the start time is adjusted to be a multiple of `step`, to avoid so-called "jumpy metrics effect"
https://github.com/kiali/kiali/blob/master/handlers/metrics.go#L85

I believe the range adjustment that takes place here should be performed _before_ the former.

Reply:
Right... You had the same comment as @jshaughn. I will move the range adjustment to the extractMetricsQueryParams.

But I have the same observation: this lead to mixing some small "business logic" in a part of the code that I see belonging to the "presentation tier".

And yes, you are right in that the adjustment of the start time to be a multiple of `step` should happen after.

Reply:
I see the problem, for lists / details queries we've introduced the business layers that sits between handlers and the underlying clients. I don't know how easy it would be to move some graph code in this layer, but maybe it's the solution? (If yes, I let you judge if it has to be part of a bigger refactoring task or not...)